File: app/src/server/models/Vocabulary.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const VocabularySchema = new Schema({
    englishWord: { type: String, required: true },
    vietnameseWord: { type: String, required: true },
    englishMeaning: { type: String, required: true },
    vietnameseMeaning: { type: String, required: true },
    topicId : { type: Schema.Types.ObjectId, ref: 'Topic', required: true },
    vocabularyStatisticId : [{ type: Schema.Types.ObjectId, ref: 'VocabularyStatistic' }],
    bookmarkVocabularyId : [{ type: Schema.Types.ObjectId, ref: 'BookmarkVocabulary' }],
});

const Vocabulary = mongoose.model('Vocabulary', VocabularySchema);
module.exports = Vocabulary;

File: app/src/main/java/com/tdtu/finalproject/model/Vocabulary.kt
package com.tdtu.finalproject.model

data class Vocabulary (
    var id: String?,
    var englishWord: String,
    var vietnameseWord: String,
    var englishMeaning: String,
    var vietnameseMeaning: String,
    var topicId: String?,
    var vocabularyStatisticId: String?,
    var bookmarkVocabularyId: String?
)